apps/src/templates/census2017/CensusTeacherBanner.jsx

```diff
@@ -0,0 +1,350 @@
+import React, {Component, PropTypes} from 'react';
+import Button from '../Button';
+import SchoolInfoInputs from '../SchoolInfoInputs';
+import styleConstants from '../../styleConstants';
+import color from "@cdo/apps/util/color";
+import {pegasus} from '@cdo/apps/lib/util/urlHelpers';
+
+const styles = {
+  button: {
+    marginLeft: 7,
+    marginRight: 7,
+    marginTop: 15,
+    marginBottom: 15,
+  },
+  buttonDiv: {
+    textAlign: 'center'
+  },
+  error: {
+    color: color.red,
+  },
+  header: {
+    marginTop: 10,
+    marginBottom: 10,
+    marginLeft: 20,
+    marginRight: 20,
+  },
+  introQuestion: {
+    marginBottom: 5
+  },
+  main: {
+    borderWidth: 1,
+    borderStyle: 'solid',
+    borderColor: color.teal,
+    minHeight: 72,
+    width: styleConstants['content-width'],
+    backgroundColor: color.white,
+    overflowWrap: 'break-word'
+  },
+  message: {
+    marginTop: 0,
+    marginBottom: 20,
+    marginLeft: 20,
+    marginRight: 20,
+  },
+  radio: {
+    verticalAlign: 'top',
+    marginRight: 10,
+  },
+  title: {
+    marginBottom: 0,
+  },
+  updateSchool: {
+    fontSize: '85%',
+    marginTop: 0,
+    marginBottom: 0,
+  },
+};
+
+export default class CensusTeacherBanner extends Component {
+  static propTypes = {
+    schoolYear: PropTypes.number.isRequired,
+    onSubmit: PropTypes.func.isRequired,
+    onDismiss: PropTypes.func.isRequired,
+    onPostpone: PropTypes.func.isRequired,
+    onChange: PropTypes.func.isRequired,
+    ncesSchoolId: PropTypes.string.isRequired,
+    question: PropTypes.oneOf(['how_many_10_hours', 'how_many_20_hours']).isRequired,
+    selection: PropTypes.bool,
+    teacherId: PropTypes.number.isRequired,
+    teacherName: PropTypes.string.isRequired,
+    teacherEmail: PropTypes.string.isRequired,
+    showUnknownError: PropTypes.bool,
+  };
+
+  constructor() {
+    super();
+    this.bindSchoolInfoInputs = this.bindSchoolInfoInputs.bind(this);
+  }
+
+  componentDidMount() {
+    this.loadSchoolName(this.props.ncesSchoolId);
+  }
+
+  state = {
+    showSchoolInfoForm: false,
+    country: 'United States',
+    schoolType: '',
+    ncesSchoolId: null,
+    schoolName: '',
+    schoolDisplayName: null,
+    schoolState: '',
+    schoolZip: '',
+    schoolLocation: '',
+  };
+
+  onCountryChange(_, event) {
+    const newCountry = event ? event.value : '';
+    this.setState({country: newCountry});
+  }
```

Comment:
note: you can define these callbacks as arrow functions to automatically bind this and remove the need to do it explicitly later. E.g.
```jsx
onCountryChange = (_, event) => {
  const newCountry = event ? event.value : '';
  this.setState({country: newCountry});
}
```

Also, nit: I prefer the callback supplied to an `onVerb` handler to be called `handleVerb` (rather than again `onVerb`), like in the react [examples](https://reactjs.org/docs/handling-events.html).

Reply:
I'll change this.